### DISCUSSION TITLE
CSS duplication without the use of additionalData config option

### DISCUSSION THREAD

**Discussion by urbik** [nuxt/nuxt#15915] [Questions]
Hey, I'm not sure if it's a bug on my side or Nuxt/Vite so I wanna first ask here.

The problem is that some of my styles are duplicated in html/css using both `nuxt build` and `nuxt dev`.

The CSS is duplicated in linked entry/default/index file when using build and when using dev the CSS is directly duplicated in html style tag.

If I remember correctly the dev variant didn't have this issue before, only the build variant had duplicated css.

In our project there are some config/vars/mixins files which we then include in our scss files when we need them. We have scss file for each page and component which we include in style section of the Vue page/component.

Do you have any idea what's wrong? It would be helpful, thx.

`"nuxt": "3.0.0-rc.11",
"sass": "^1.55.0",`

import { defineNuxtConfig } from 'nuxt/config';
import eslintPlugin from 'vite-plugin-eslint';

export default defineNuxtConfig({
  vite: {
    plugins: [
      eslintPlugin(),
    ],
  },

  alias: {
    '@images': './assets/images',
    '@data': './assets/other/data',
  },

  modules: [
    '@vueuse/nuxt',
    'vite-plugin-vue-type-imports/nuxt',
  ],
});

For eaxample the :root variables are duplicated several times:
![image](https://user-images.githubusercontent.com/10115217/205684060-be9c7088-aca4-4491-9550-1cd47ead1334.png)

Button css is also duplicated:
![image](https://user-images.githubusercontent.com/10115217/205684694-427e0995-7e5f-4b3f-8865-2a7fa5a7821c.png)

**Comment by TheAlexLichter** [comment]
Hey! Could you kindly provide a [reproduction](https://v3.nuxtjs.org/community/reporting-bugs/#create-a-minimal-reproduction)? 🙏

**Reply by urbik** [reply]
Ok, I will soon.

**Reply by urbik** [reply]
Ok, got it. It seems we have inour project 2 errors. One with nuxt dev and one with nuxt build. The one with nuxt dev is probably caused by a bit of a mess in files and imports but I was able to reproduce the problem with nuxt build we have now and had from the start of the project. You can see it here:

![image](https://user-images.githubusercontent.com/10115217/205939696-1798f397-c7b5-4471-83c3-9d05bbc66ba2.png)

And here is a stackblitz project: https://stackblitz.com/edit/urbik-css-duplication-on-nuxt-build

Just run yarn build and then run node using command showing in terminal to access the app in the preview window.

Anyone has a clue, please?

**Comment by urbik** [comment]
I was working on rearanging my project's (s)css files and got to a point i solved my new issues but the old one, the one I'm asking about here is still unresolved! One of the changes I made was that i moved my scss files from assets folder to public folder. I will look at dependencies tommorrow and if it won't be caused by some of them then it may be a significant issue with nuxt/vite. Can somebody look into it too, please? @manniL

**Comment by urbik** [comment]
Well, I updated package.json to a minimal dependencies and it still duplicates CSS on build... @manniL or anyone?
https://stackblitz.com/edit/urbik-css-duplication-on-nuxt-build?file=pages/index.vue

![image](https://user-images.githubusercontent.com/10115217/206468760-4fbcd390-39d4-426c-83a5-0e9b5d2c5188.png)

**Comment by brunokunace** [comment]
same problem here, i have 5x the same css =/

**Comment by RayJason** [comment]
After build, I found two similar css files: entry.css and assets.css  

This css is from  '@/assets/css/element.override.css':  

![devtool](https://github.com/nuxt/nuxt/assets/15711610/fd4507e7-245f-4514-9e89-14cfee96fbf7)

I want to know who generated these two files respectively.  
Their difference is:
1. `entry.*.css` 54kb / `asset.*.css` 44kb
2. They both package Tailwind, but in different locations  


![image](https://github.com/nuxt/nuxt/assets/15711610/ddaacdc3-369d-42e0-a704-4ac8c873d0c0)


Here is my nuxt.config file:  
```ts
export default defineNuxtConfig({
  devServer: {
    port: 3050,
  },
  typescript: {
    typeCheck: true,
  },
  modules: [
    '@pinia/nuxt',
    '@element-plus/nuxt',
    '@vueuse/nuxt',
    '@nuxtjs/tailwindcss',
    '@nuxtjs/eslint-module',
    '@pinia-plugin-persistedstate/nuxt',
    'nuxt-lodash',
    'nuxt-icons',
    'dayjs-nuxt',
  ],
  extends: ['nuxt-seo-kit'],
  css: [
    '@/assets/css/global.css',
    '@/assets/css/mainsite.css',
    '@/assets/css/element.override.css',
    '@/assets/css/font.css',
  ],
  runtimeConfig: {
    public: {
      siteUrl: process.env.SITE_URL,
      version: pkg.version,
    },
  },
  imports: {
    dirs: ['./stores'],
  },
  pinia: {
    autoImports: ['defineStore', 'acceptHMRUpdate'],
  },
  linkChecker: {
    failOn404: false,
    exclude: ['/**'],
  },
  vite: {
    vue: {
      script: {
        defineModel: true,
        propsDestructure: true,
      },
    },
  },
})
```

**Comment by DraftProducts** [comment]
I have the same issue with nuxt 3.10.3.
![image](https://github.com/nuxt/nuxt/assets/29425008/d8883f5c-05b9-4bb6-9f3a-f0184378dedd)

**Comment by kamenivskyi** [comment]
It looks like this is a problem with inline styles. Try to disable it in nuxt.config
  ```
  features: {
        inlineStyles: false,
    },
```
